source/Nabla/COpenGLDriver.cpp

```diff
@@ -1259,34 +1259,42 @@ core::smart_refctd_ptr<IGPUSpecializedShader> COpenGLDriver::createGPUSpecialize
     const std::string& EP = _specInfo.entryPoint;
     const asset::ISpecializedShader::E_SHADER_STAGE stage = _specInfo.shaderStage;
 
-    core::smart_refctd_ptr<asset::ICPUShader> spvCPUShader = nullptr;
+    core::smart_refctd_ptr<asset::ICPUBuffer> spirv;
     if (glUnspec->containsGLSL())
     {
         auto begin = reinterpret_cast<const char*>(glUnspec->getSPVorGLSL()->getPointer());
         auto end = begin+glUnspec->getSPVorGLSL()->getSize();
         std::string glsl(begin,end);
-        asset::ICPUShader::insertGLSLExtensionsDefines(glsl, getSupportedGLSLExtensions().get());
+        COpenGLShader::insertGLtoVKextensionsMapping(glsl, getSupportedGLSLExtensions().get());
         auto glslShader_woIncludes = GLSLCompiler->resolveIncludeDirectives(glsl.c_str(), stage, _specInfo.m_filePathHint.c_str());
         //{
         //    auto fl = fopen("shader.glsl", "w");
         //    fwrite(reinterpret_cast<const char*>(glslShader_woIncludes->getSPVorGLSL()->getPointer()), 1, glslShader_woIncludes->getSPVorGLSL()->getSize(), fl);
         //    fclose(fl);
         //}
-        core::smart_refctd_ptr<asset::ICPUBuffer> spvCode = GLSLCompiler->compileSPIRVFromGLSL(
+        spirv = GLSLCompiler->compileSPIRVFromGLSL(
                 reinterpret_cast<const char*>(glslShader_woIncludes->getSPVorGLSL()->getPointer()),
                 stage,
                 EP.c_str(),
                _specInfo.m_filePathHint.c_str()
             );
 
-        if (!spvCode)
+        if (!spirv)
             return nullptr;
-
-        spvCPUShader = core::make_smart_refctd_ptr<asset::ICPUShader>(std::move(spvCode));
     }
     else
-        spvCPUShader = core::make_smart_refctd_ptr<asset::ICPUShader>(core::smart_refctd_ptr<asset::ICPUBuffer>(glUnspec->m_code));
+    {
+        spirv = glUnspec->m_code;
+    }
+#ifndef _IRR_DEBUG
+    if (SPIRVOptimizer)
+        spirv = SPIRVOptimizer->optimize(spirv.get());
+#endif
```

Comment:
this is just weird, dont do this sort of crap where debug doesn't optimize GLSL ever.

Reply:
also why is the built-in optimizer `SPIRVOptimizer` used always, why not allow optimizer as argument and use `SPIRVOptimizer` as a default for that argument ?